Emby.Server.Implementations/Services/SwaggerService.cs

```diff
@@ -0,0 +1,259 @@
+        public SwaggerComponents components { get; set; }
+    }
+
+    public class SwaggerComponents
+    {
+        public Dictionary<string, SwaggerSecurityScheme> securitySchemes { get; set; }
+    }
+
+    public class SwaggerSecurityScheme
+    {
+        public string name { get; set; }
+        public string type { get; set; }
+        public string @in { get; set; }
+    }
+
+    public class SwaggerInfo
+    {
+        public string description { get; set; }
+        public string version { get; set; }
+        public string title { get; set; }
+        public string termsOfService { get; set; }
+
+        public SwaggerConcactInfo contact { get; set; }
+    }
+
+    public class SwaggerConcactInfo
+    {
+        public string email { get; set; }
+        public string name { get; set; }
+        public string url { get; set; }
+    }
+
+    public class SwaggerTag
+    {
+        public string description { get; set; }
+        public string name { get; set; }
+    }
+
+    public class SwaggerMethod
+    {
+        public string summary { get; set; }
+        public string description { get; set; }
+        public string[] tags { get; set; }
+        public string operationId { get; set; }
+        public string[] consumes { get; set; }
+        public string[] produces { get; set; }
+        public SwaggerParam[] parameters { get; set; }
+        public Dictionary<string, SwaggerResponse> responses { get; set; }
+        public Dictionary<string, string[]>[] security { get; set; }
+    }
+
+    public class SwaggerParam
+    {
+        public string @in { get; set; }
+        public string name { get; set; }
+        public string description { get; set; }
+        public bool required { get; set; }
+        public string type { get; set; }
+        public string collectionFormat { get; set; }
+    }
+
+    public class SwaggerResponse
+    {
+        public string description { get; set; }
+
+        // ex. "$ref":"#/definitions/Pet"
+        public Dictionary<string, string> schema { get; set; }
+    }
+
+    public class SwaggerDefinition
+    {
+        public string type { get; set; }
+        public Dictionary<string, SwaggerProperty> properties { get; set; }
+    }
+
+    public class SwaggerProperty
+    {
+        public string type { get; set; }
+        public string format { get; set; }
+        public string description { get; set; }
+        public string[] @enum { get; set; }
+        public string @default { get; set; }
+    }
+
+    public class SwaggerService : IService, IRequiresRequest
+    {
+        private SwaggerSpec _spec;
+
+        public IRequest Request { get; set; }
+
+        public object Get(GetSwaggerSpec request)
+        {
+            return _spec ?? (_spec = GetSpec());
+        }
+
+        private SwaggerSpec GetSpec()
+        {
+            string host = null;
+            Uri uri;
+            if (Uri.TryCreate(Request.RawUrl, UriKind.Absolute, out uri))
```

Comment:
```suggestion
            if (Uri.TryCreate(Request.RawUrl, UriKind.Absolute, out Uri uri))
```
Oneliners :)

Reply:
@Bond-009 I'll be rewriting this to support OpenAPI spec 3.0. Style changes will be addressed then.